server/src/main/java/org/elasticsearch/action/search/FetchSearchPhase.java

```diff
@@ -73,6 +73,7 @@
         this.context = context;
         this.logger = context.getLogger();
         this.resultConsumer = resultConsumer;
+        this.context.getTask().getStatus().phaseStarted(getName(), context.getNumShards());
```

Comment:
I am pretty positive that I am not counting fetch operations properly. Not sure where I should get the right count though. I've seen that for instance, if I have two shards, one is skipped and the other one is queried, fetch now still reports fetch operations from two shards but that is incorrect.

Reply:
It depends what we want to report here. I would start from the number of non-null values in [`docIdsToLoad`](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/search/FetchSearchPhase.java#L113) since that's the number of fetch task that we expect. The other shards are ignored so it would be weird to report 10 shards for the phase but fewer tasks since we don't run the fetch phase on shards that don't have a result in the global top hits.

Reply:
good point. It looks like though you cannot know in advance the number of shards you are going to send requests to, not before starting to execute the phase. That is a bit annoying, I would prefer to know as soon as the phase is initialized, as the number of shards is currently the second argument of `phaseStarted`.